nav2_docking/opennav_docking/src/simple_charging_dock.cpp

```diff
@@ -323,6 +356,118 @@ void SimpleChargingDock::jointStateCallback(const sensor_msgs::msg::JointState::
   is_stalled_ = (velocity < stall_velocity_threshold_) && (effort > stall_effort_threshold_);
 }
 
+bool SimpleChargingDock::startDetectionProcess()
+{
+  // Skip if already starting or ON
+  if (detection_started_) {
+    return true;
+  }
+
+  // 1. Service START request
+  if (detector_client_) {
+    auto req = std::make_shared<std_srvs::srv::Trigger::Request>();
+    try {
+      auto future = detector_client_->invoke(
+        req,
+        std::chrono::duration_cast<std::chrono::nanoseconds>(
+          std::chrono::duration<double>(detector_service_timeout_)));
+
+      if (!future || !future->success) {
+        RCLCPP_ERROR(
+          node_->get_logger(), "Detector service '%s' failed to start.",
+          detector_service_name_.c_str());
+        return false;
+      }
+    } catch (const std::exception & e) {
+      RCLCPP_ERROR(
+        node_->get_logger(), "Calling detector service '%s' failed: %s",
+        detector_service_name_.c_str(), e.what());
+      return false;
+    }
+  }
+
+  // 2. Subscription toggle
+  //    Only subscribe once; will set state to ON on first message
+  if (subscribe_toggle_ && !dock_pose_sub_) {
```

Comment:
I'm not directly asking you to do this, just a point of discussion:

So in the startup of the sensor, we could, in concept, have a while loop to spin/block for the perception timeout, then return true/false if we obtain data or not. That would remove the `initial_pose_received_` variable since we wouldn't continue if we didn't obtain it.